bundles/org.openhab.automation.jsscripting/src/main/java/org/openhab/automation/jsscripting/internal/scope/AbstractScriptExtensionProvider.java

```diff
@@ -41,7 +45,6 @@ protected void addType(String name, Function<String, Object> value) {
 
     @Activate
     public void activate(final BundleContext context) {
-        types = new HashMap<>();
         initializeTypes(context);
```

Comment:
Too lazy/busy to check right now: Will `initializeTypes` clear the HashMap, so it will be okay if it already has previous contents if `activate` would be called again?

Reply:
Difficult to say. There is no class extending this class. I would vote for `types.clear();` before initializing.

Reply:
Done in https://github.com/openhab/openhab-addons/commit/e17f869c4e8b6459cfbf1510e3117a0c4745f2d2.